MovieQuiz/Services/MoviesLoader.swift

```diff
@@ -0,0 +1,42 @@
+
+import Foundation
+
+protocol MoviesLoading {
+    func loadMovies(handler: @escaping (Result<MostPopularMovies, Error>) -> Void)
+}
+
+struct MoviesLoader: MoviesLoading {
+    // MARK: - NetworkClient
+    private let networkClient = NetworkClient()
+    
+    // MARK: - URL
+    private var mostPopularMoviesUrl: URL {
+        // Если мы не смогли преобразовать строку в URL, то приложение упадёт с ошибкой
+        guard let url = URL(string: "https://imdb-api.com/en/API/Top250Movies/k_jc9jdg1b") else {
+            preconditionFailure("Unable to construct mostPopularMoviesUrl")
+        }
+        return url
+    }
+    
+    private enum DecodeError: Error {
+        case decodeError
+    }
+    
+    func loadMovies(handler: @escaping (Result<MostPopularMovies, Error>) -> Void) {
+        networkClient.fetch(url: mostPopularMoviesUrl, handler: { result in
+            switch result {
+                case .failure(let error):
+                    handler(.failure(error))
+                
+                case .success(let data):
+                    let JSONtoMovieList = try? JSONDecoder().decode(MostPopularMovies.self, from: data)
```

Comment:
JSONtoMovieList – c именованием переменных, обычно принято начинать с нижнего camelCase. То есть jsonToMovieList - смотрелось бы чище и правильнее, потому что новое слово To - начинается с верхнего регистра.
Но так как JSON - и аббревиатура тоже, то ей положено быть в CAPS'e :) 

поэтому я бы такое именование избегала. Давай посмотрим, что будет результатом функции decode? Объект, какой? MostPopularMovies! так вполне подойдет
```suggestion
                    let mostPopularMovies = try? JSONDecoder().decode(MostPopularMovies.self, from: data)
```